stun/stun.go

```diff
@@ -152,14 +155,14 @@ func doStunTransaction(lAddr, rAddr *net.UDPAddr) (publicIP net.IP, publicPort i
 
 	privIp := net.ParseIP(lIP)
 	defer func() {
-		if !privIp.Equal(publicIP) {
+		if publicIP != nil && privIp != nil && !privIp.Equal(publicIP) {
```

Comment:
will privIP be ever nil?

Reply:
I don't think so.   the privIP is always there in the test.  It's a strict check in the logic.  Wanna remove it?